.github/workflows/rust.yml

```diff
@@ -85,94 +76,40 @@ jobs:
         uses: actions-rs/toolchain@v1
         with:
           toolchain: 1.53.0
-          components: clippy, rustfmt
           default: true
 
-      - name: Build binary
-        run: |
-          cargo build --verbose --release
-      - name: Run tests
+      - name: "[Linux ONLY] Install dependencies"
+        if: ${{ matrix.os == 'ubuntu-latest' }}
         run: |
-          cargo test --verbose
-
-      - name: Run clippy
-        uses: actions-rs/clippy-check@v1.0.7
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          args: --all-targets -- -D warnings
-      - name: Check formatting
-        run: |
-          cargo fmt --all -- --check
-
-      - name: Upload binary
-        uses: actions/upload-artifact@v2
-        with:
-          name: leafish.exe
-          path: target/release/leafish.exe
-      - name: Release binary
-        if: startsWith(github.ref, 'refs/tags/')
-        uses: softprops/action-gh-release@v1
-        with:
-          files: |
-            target/release/leafish.exe
-
-  macos:
-    runs-on: macos-11
-
-    steps:
-      - uses: actions/checkout@v2
-
-      - uses: actions/cache@v2
-        with:
-          path: |
-            ~/.cargo/bin/
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-            target/
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
-
-      - name: Install Rust
-        uses: actions-rs/toolchain@v1
-        with:
-          toolchain: 1.53.0
-          components: clippy, rustfmt
-          default: true
+          sudo apt-get update
+          sudo apt-get install -y --no-install-recommends libxcb-shape0-dev libxcb-xfixes0-dev libxkbcommon-dev
 
       - name: Build binary
-        run: |
-          cargo build --verbose --release
-          chmod a+x target/release/leafish
         env:
           MACOSX_DEPLOYMENT_TARGET: 10.14
+        run: cargo build --verbose --release
       - name: Run tests
-        run: |
-          cargo test --verbose
+        run: cargo test --verbose
 
-      - name: Run clippy
-        uses: actions-rs/clippy-check@v1.0.7
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          args: --all-targets -- -D warnings
-      - name: Check formatting
-        run: |
-          cargo fmt --all -- --check
-
-      - name: Package binary
+      - name: "[MacOS ONLY] Package binary"
+        if: ${{ matrix == 'macos-11' }}
         run: |
+          chmod a+x target/release/leafish
```

Comment:
Since you're adding this here (which makes sense), it should probably be removed from line 100